src/connection_manager.c

```diff
@@ -254,9 +252,8 @@ qd_config_ssl_profile_t *qd_dispatch_configure_ssl_profile(qd_dispatch_t *qd, qd
     ssl_profile->ssl_trusted_certificates   = qd_entity_opt_string(entity, "trustedCerts", 0); CHECK();
     ssl_profile->ssl_uid_format             = qd_entity_opt_string(entity, "uidFormat", 0); CHECK();
     ssl_profile->ssl_display_name_file      = qd_entity_opt_string(entity, "displayNameFile", 0); CHECK();
-    sys_mutex_lock(qd->connection_manager->ssl_profile_lock);
-    ssl_profile->ref_count                  = 0;
-    sys_mutex_unlock(qd->connection_manager->ssl_profile_lock);
+
+    __sync_and_and_fetch(&ssl_profile->ref_count, 0);
```

Comment:
No, i didn't find an atomic assign, but come to think of it, i think just regular assignment should work. This was just a workaround I did without thinking.